pysph/cpy/low_level.py

```diff
@@ -172,21 +174,28 @@ def _generate(self):
 
         self.tp.compile()
         self.source = self.tp.source
-        self.knl = getattr(self.tp.mod, self.name)
-        import pyopencl as cl
-        self._max_work_group_size = self.knl.get_work_group_info(
-            cl.kernel_work_group_info.WORK_GROUP_SIZE,
-            self.queue.device
-        )
+
+        if self.backend == 'opencl':
+            self.knl = getattr(self.tp.mod, self.name)
+            import pyopencl as cl
+            self._max_work_group_size = self.knl.get_work_group_info(
+                cl.kernel_work_group_info.WORK_GROUP_SIZE,
+                self.queue.device
+            )
+        elif self.backend == 'cuda':
+            self.knl = self.tp.mod.get_function(self.name)
 
     def _correct_opencl_address_space(self):
         code = self.tp.blocks[-1].code.splitlines()
-        code[0] = 'KERNEL ' + code[0]
+        code[0] = 'KERNEL ' + code[0][13:]
```

Comment:
Am a bit confused why the [13:] needed, can you just add a comment explaining this?  I assume it is to remove `'WITHIN_KERNEL'`.